app/main.py

```diff
@@ -2,5 +2,14 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    cashe = {}
+
+    def wrapper(*args) -> Callable:
+        if args in cashe:
+            print("Getting from cache")
+            return cashe[args]
+        print("Calculating new result")
+        cashe[args] = func(*args)
```

Comment:
Your approach is incorrect: you should store arguments and the result of the function with these arguments.